File: cf 918 div 4/e.cpp
#include <iostream>
#include <unordered_map>
using namespace std;

int main() {
    int t;
    cin >> t;

    while (t--) {
        int n;
        cin >> n;
        vector<int> arr(n);

        for (int i = 0; i < n; i++) {
            cin >> arr[i];
            if (i % 2 == 0) {
                arr[i] = -arr[i];
            }
        }

        int s = 0;
        unordered_map<int, bool> pres;
        bool f = false;

        for (int i = 0; i < n; i++) {
            s += arr[i];
            if (pres.find(s) != pres.end()) {
                f = true;
                break;
            } else {
                pres[s] = true;
            }
        }

        if (f) {
            cout << "YES\n";
        } else {
            cout << "NO\n";
        }
    }

    return 0;
}

File: cf 918 div 4/e.py
from sys import stdin
def input(): return stdin.readline().rstrip("\r\n")


for _ in range(int(input())):
    n=int(input())
    arr=list(map(int,input().split()))
    for i in range(0,n,2):arr[i]=-arr[i]
    s=0;pres={0};f=False

    for ele in arr:
        s+=ele
        if s in pres:f=True
        else:pres.add(s)

        if f:break
    if f:print('YES')
    else:print('NO')
